File: Lab 4/C++/LetterProgram.cpp
#include "LetterHeader.h"
#include <algorithm>

using namespace Artem;

LetterString::LetterString() : StringBegin() {} // Виклик конструктора за замовчуванням базового класу

LetterString::LetterString(const std::string& s) : StringBegin(s) {} // Виклик конструктора з параметром базового класу

LetterString::LetterString(const LetterString& other) : StringBegin(other) {} // Конструктор копіювання



// Сортування 

void LetterString::sortString()
{
    std::string row = getString(); // Отримуємо рядок
    std::sort(row.begin(), row.end()); // Сортуємо
    setString(row); // Оновлюємо через protected метод
}

File: Lab 4/C#/LetterProgram.cs
using System;
using Artem;

public class LetterString : StringBegin
{
    //(явний виклик базового конструктора)
    public LetterString(string value) : base(value) { }

    public void Sort_Str()
    {
        string row = GetValue(); // Отримуємо рядок
        char[] arr = row.ToCharArray(); // Перетворюємо на масив символів
        Array.Sort(arr); // Сортуємо масив
        SetValue(new string(arr));
    }
}
